### PR TITLE
add data sync

### PR DESCRIPTION
**Pull request by wei-lee**
## Motivation

Add data sync as a new service

### CODE REVIEW

**Comment by wei-lee** [comment]
note: we should merge this PR and publish it as part of a new community release.

File: _data/services.yml [hunk 1]
```diff
@@ -1,3 +1,19 @@
...
+  icon: sync
+  link: /services/data-sync/
+  description: Add realtime data synchronization and offline capabilities to your mobile app
+  features:
+    - Use GraphQL to design the APIs for your mobile app
+    - Add real time data synchronisation to your mobile app
+    - Allow your moible app to work offline
+    - Easily resolve conflicts with the flexible conflict resolution strategies
+    - Support authentication and authorization using the Identity Management service or choice of your own
+    - Great observability through the Mobile Metrics Service
```
Comment (finp) [inline _data/services.yml]: AeroGear Mobile Metrics Service

File: sdks/index.html [hunk 2]
```diff
@@ -50,6 +56,8 @@
           <a href="https://docs.aerogear.org/aerogear/latest/mobile-developer-console.html#android">Android
```
Comment (laurafitzgerald) [inline sdks/index.html]: It's not part of this change but I noticed the links for the SDK labels in the table don't work correctly. Each one navigates to the top Section of Mobile Developer Console Page.

File: _includes/homepage-header.html [hunk 3]
```diff
@@ -4,15 +4,12 @@ <h1>Cloud Native Mobile App Development on OpenShift</h1>
...
-    <div class="col-md-7 tagline">
+    <div class="col-sm-12">
       <h3> We make enterprise mobile development easy</h3>
       <p>AeroGear provides a suite of Mobile Services across a range of native and hybrid client SDKs that will simplify your mobile and modern application development needs, leveraging OpenShift's Container technology to provide a secure, scalable backend platform.</p>
       <div class="main-action">
         <a href="/getting-started/" class="btn btn-accent-inverse btn-lg">Get started now</a>
       </div>
     </div>
-    <div class="col-md-5 video">
-      <iframe width="100%" height="100%" src="https://www.youtube.com/embed/OuMWnx-FS3Q" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe>
```
Comment (jasonmadigan) [inline _includes/homepage-header.html]: Poor John

Comment (laurafitzgerald) [reply _includes/homepage-header.html]: Would he make us another video with new screengrabs? 😄

File: _data/services.yml [hunk 4]
```diff
@@ -1,3 +1,19 @@
+sync:
+  name: Data Sync
+  icon: sync
+  link: /services/data-sync/
+  description: Add realtime data synchronization and offline capabilities to your mobile app
+  features:
+    - Use GraphQL to design the APIs for your mobile app
```
Comment (wtrocki) [inline _data/services.yml]: I think that usually on those websites statements are passive. For example:

`Enables real time updates` `Leverages GraphQL`

File: _data/services.yml [hunk 5]
```diff
@@ -1,3 +1,19 @@
...
+    - Add real time data synchronisation to your mobile app
+    - Allow your moible app to work offline
```
Comment (wtrocki) [inline _data/services.yml]: Typo. I think mobile app sounds too vague and may be confusing for native felas. If target is cordova and web then we should specify that. 

The main target for this is to give plain facts about sync.

- Works with web and cordova
- Uses GraphQL (that will imply many features out of the box)

File: _data/services.yml (hunk 4 above)
Comment (wei-lee) [reply _data/services.yml]: I was following the same format as the features for other services. If we change it here, we have to change all of them.

File: _data/services.yml (hunk 5 above)
Comment (wei-lee) [reply _data/services.yml]: the sdks sections clearly indicates what sdks are available. For this section, I think a bit high level is ok.

File: _data/services.yml [hunk 6]
```diff
@@ -1,3 +1,19 @@
...
+    - Easily resolve conflicts with the flexible conflict resolution strategies
+    - Support authentication and authorization using the Identity Management service or choice of your own
+    - Great observability through the Mobile Metrics Service
+    - Track all actions with audit logs
+  documentation: https://docs.aerogear.org/aerogear/latest/data-sync.html
```
[the remark this answers is not in the document]
Comment (wei-lee) [reply _data/services.yml]: I think it's ok. It's the same as all the other services in this page.

File: sdks/index.html (hunk 2 above)
Comment (wei-lee) [reply sdks/index.html]: I don't really know where should they link to TBH, I think I will leave it for the moment.

File: _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+## Try It Out
+
+If you like to try out the new features, please follow our [getting-started guide](https://docs.aerogear.org/aerogear/latest/getting-started.html) with the new [Cordova showcase app](https://github.com/aerogear/ionic-showcase). Please remember, feedbacks are always welcome and we can't wait to hear what you think!
+
+Thanks.
+
+\- *The AeroGear Team*
+
+
```
Comment (wei-lee) [inline _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md]: @finp do you mind quickly go through this blogpost and see if what do you think?

File: _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+* A new [Cordova showcase app](https://github.com/aerogear/ionic-showcase) is created to demonstrate the new data sync capabilities and how to use it with other services.
+* The APIs of the [AeroGear JS SDK](https://github.com/aerogear/aerogear-js-sdk) core module has been refactored to make it easier for developers to consume.
+* A new [installer repo](https://github.com/aerogear/mobile-services-installer) is created to make it easier for developers to install AeroGear Mobile Services. You can also find out what versions of each service is included in a given release. For example, the 2.0.0 release contains [these services](https://github.com/aerogear/mobile-services-installer/blob/2.0.0/versions.yml).
```
Comment (finp) [inline _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md]:
```suggestion
* A new [installer repo](https://github.com/aerogear/mobile-services-installer) is created to make it easier for developers to install AeroGear Mobile Services. You can also find out which versions of each service are included in a given release. For example, the 2.0.0 release contains [these services](https://github.com/aerogear/mobile-services-installer/blob/2.0.0/versions.yml).
```

File: _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+title: AeroGear Mobile Services version 2.0.0 is here
+author: aerogear
+tag: community
+---
+
+About 7 months ago, we announced [the 1.0.0 release](/news/2018/07/04/aerogear-mobile-services-announcement/index.html) of the AeroGear Mobile Services. Since then we have been working very hard to continue the development work. Today we are delighted to announce the availability of AeroGear Mobile Services version 2.0.0.
+
+## What's New
+
+* A new [Data Sync](/services/data-sync/) framework is now available. It is built on top of the popular [Apollo GraphQL framework](https://www.apollographql.com/). We added a set of new features like offline support, conflict handling etc to make it easier for developers to create mobile applications that need to sync data between clients and server. To learn more, please checkout the [Voyager server](https://github.com/aerogear/voyager-server) and the [Voyager client](https://github.com/aerogear/aerogear-js-sdk/tree/master/packages/sync) repos.
```
Comment (wtrocki) [inline _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md]: Minor suggestion `checkout` => `check`

File: _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+* A new [Cordova showcase app](https://github.com/aerogear/ionic-showcase) is created to demonstrate the new data sync capabilities and how to use it with other services.
```
Comment (StephenCoady) [inline _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md]:
```suggestion
* A new [Cordova showcase app](https://github.com/aerogear/ionic-showcase) is created to demonstrate the new data sync capabilities and how to use it with other services.
```
* A new [Cordova showcase app](https://github.com/aerogear/ionic-showcase) was created to demonstrate the new data sync capabilities and how to use it with other services.

File: _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+* The APIs of the [AeroGear JS SDK](https://github.com/aerogear/aerogear-js-sdk) core module has been refactored to make it easier for developers to consume.
+* A new [installer repo](https://github.com/aerogear/mobile-services-installer) is created to make it easier for developers to install AeroGear Mobile Services. You can also find out what versions of each service is included in a given release. For example, the 2.0.0 release contains [these services](https://github.com/aerogear/mobile-services-installer/blob/2.0.0/versions.yml).
+* Bug fixes and improvements for the exiting services.
+
+
+## Try It Out
+
+If you like to try out the new features, please follow our [getting-started guide](https://docs.aerogear.org/aerogear/latest/getting-started.html) with the new [Cordova showcase app](https://github.com/aerogear/ionic-showcase). Please remember, feedbacks are always welcome and we can't wait to hear what you think!
```
Comment (finp) [inline _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md]: should we link to https://docs.aerogear.org/aerogear/latest/data-sync.html ?

File: _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+
+## What's New
+
+* A new [Data Sync](/services/data-sync/) framework is now available. It is built on top of the popular [Apollo GraphQL framework](https://www.apollographql.com/). We added a set of new features like offline support, conflict handling etc to make it easier for developers to create mobile applications that need to sync data between clients and server. To learn more, please checkout the [Voyager server](https://github.com/aerogear/voyager-server) and the [Voyager client](https://github.com/aerogear/aerogear-js-sdk/tree/master/packages/sync) repos.
+* A new [Cordova showcase app](https://github.com/aerogear/ionic-showcase) is created to demonstrate the new data sync capabilities and how to use it with other services.
+* The APIs of the [AeroGear JS SDK](https://github.com/aerogear/aerogear-js-sdk) core module has been refactored to make it easier for developers to consume.
```
Comment (StephenCoady) [inline _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md]:
```suggestion
* The APIs of the [AeroGear JS SDK](https://github.com/aerogear/aerogear-js-sdk) core module have been refactored to make it easier for developers to consume.
```

File: _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md (hunk 8 above)
Comment (StephenCoady) [inline _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md]:
```suggestion
* A new [installer repo](https://github.com/aerogear/mobile-services-installer) was created to make it easier for developers to install AeroGear Mobile Services. You can also find out what versions of each service is included in a given release. For example, the 2.0.0 release contains [these services](https://github.com/aerogear/mobile-services-installer/blob/2.0.0/versions.yml).
```

File: _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+* A new [installer repo](https://github.com/aerogear/mobile-services-installer) is created to make it easier for developers to install AeroGear Mobile Services. You can also find out what versions of each service is included in a given release. For example, the 2.0.0 release contains [these services](https://github.com/aerogear/mobile-services-installer/blob/2.0.0/versions.yml).
+* Bug fixes and improvements for the exiting services.
```
Comment (StephenCoady) [inline _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md]:
```suggestion
* Bug fixes and improvements for the existing services.
```

File: _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md (hunk 11 above)
Comment (StephenCoady) [inline _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md]:
```suggestion
If you would like to try out the new features, please follow our [getting-started guide](https://docs.aerogear.org/aerogear/latest/getting-started.html) with the new [Cordova showcase app](https://github.com/aerogear/ionic-showcase). Please remember, feedback is always welcome and we can't wait to hear what you think!
```

File: _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (wei-lee) [reply _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md]: the apis are already versioned. I am not sure how they are generated though. @wtrocki any idea?

File: _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md (hunk 11 above)
Comment (wei-lee) [reply _posts/2019-03-22-aerogear-mobile-services-2.0.0-release.md]: yes, I will add it here.

**Comment by wei-lee** [comment]
@finp @laurafitzgerald mind approving the PR before I merge?
